src/exports.c

```diff
@@ -22,24 +22,40 @@ typedef struct Needles {
     void          (*free) (void **);
     uint32_t       *leapyears;
     uint32_t        length;
-    void           *result;
+    void           *result; // output
+    bool          (*action) (bool, json_object *, Needles);
+    bool          (*filter) (int, json_object *, Needles);
```

Comment:
Not easy to understand what `int` here means (one needs to go through other parts of the code and read documentation for `json-c` first). Probably would be better to make it something like `bool found`. Not critical, but would be great for the sake of readability.

Reply:
The problem is that json-c returns false as '-1', which is a very old way of doing boolean and not compatible with stdbool.h.

I can venture to call the thing `int jp_rv` everywhere if that helps.

Reply:
I'd like the struct to be as concise as possible here, and as to bool, it's not going to happen because of json-c.

Reply:
what i meant is that you could call `filter` like that:
```c
needles[i]->filter(ret == 0, found, needles[i])
```
and then `filter` func may have clearer semantics.

Reply:
ofc it's your choice, i won't insist.

Reply:
Good points, addressed.